server/src/main/java/org/elasticsearch/index/analysis/NamedAnalyzer.java

```diff
@@ -55,6 +55,8 @@ public NamedAnalyzer(String name, AnalyzerScope scope, Analyzer analyzer) {
         this.positionIncrementGap = positionIncrementGap;
         if (analyzer instanceof org.elasticsearch.index.analysis.CustomAnalyzer) {
             this.analysisMode = ((org.elasticsearch.index.analysis.CustomAnalyzer) analyzer).getAnalysisMode();
+        } else if (analyzer instanceof org.elasticsearch.index.analysis.ReloadableCustomAnalyzer) {
+            this.analysisMode = ((org.elasticsearch.index.analysis.ReloadableCustomAnalyzer) analyzer).getAnalysisMode();
         } else {
```

Comment:
Maybe instead do an instanceof check for AnalyzerComponentsProvider and then call `getComponents().getAnalysisMode()`?

Reply:
Oh for Java pattern matching... thanks!